internal/subshell/subshell.go

```diff
@@ -188,19 +189,19 @@ func DetectShell(cfg sscommon.Configurable) (string, string) {
 	}
 
 	if binary == "" {
-		binary = detectShellParent()
+		binary = configured
```

Comment:
Sorry I realized upon reading the changes that configured had a different connotation than I remembered. In my mind this was "what the user configured", but actually this is just a fallback we ourselves set when we detect the shell.

This is awkward and should probably be revisited, but given the current behaviour I this should probably be a fallback after shell parent and environment.